### ISSUE TITLE
Question: What is a Migrator?

### ISSUE THREAD

**Issue by simbo1905** [coditory/sherlock-distributed-lock#237]
Hello 👋 

I was looking 👀 at the docs 📝 looking at https://coditory.github.io/sherlock-distributed-lock/migrator/ I am confused:

I have used many database migration tools and none of them need a distributed lock service.

What is the relationship between a distributed locking service like Sherlock and database migrations? 

Thanks!

**Comment by pmendelski** [comment]
Hi,

Thanks for submitting this question. I believe it could help others.

TLDR; Database migration tools don't require distributed locks, but they can be easily implemented on top of distributed locks. That's it.

Longer version:
You're right, database migration tool don't require a distributed lock if it can be guaranteed they are not executed in parallel. If they are executed in parallel (for example during multi instance application startup) some limiting mechanism must be applied. Even without the need for distributed lock they usually maintain a table/collection in DB to track applied migrations. Distributed locks provide both: migration marking mechanism and migration limitting.

The main benefits of using the Migrator from `sherlock-distributed-lock` library:
- Easy way to apply migrations during process startup. Thanks to using distributed locks it's ok to start migrations in parallel from multiple processes - only one will acquire the lock.
- Less dependencies and mental overhead - just reusing the same mechanism.

**Comment by simbo1905** [comment]
Thanks for that. We use GitHub Actions, which must internally be using a distributed lock. So your answer makes clear that the code shows how to write such a capability.
